File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/WebviewNativeConfigParams.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class WebviewNativeConfigParams(
  val view: ViewEnum, // Oneof: multiple, single
  val cspSource: String,
  val webviewBundleServingPrefix: String,
  val rootDir: String? = null,
  val injectScript: String? = null,
  val injectStyle: String? = null,
) {

  enum class ViewEnum {
    @SerializedName("multiple") Multiple,
    @SerializedName("single") Single,
  }
}

File: agent/bindings/csharp/WebviewNativeConfigParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class WebviewNativeConfigParams
  {
    [JsonProperty(PropertyName = "view")]
    public ViewEnum View { get; set; } // Oneof: multiple, single
    [JsonProperty(PropertyName = "cspSource")]
    public string CspSource { get; set; }
    [JsonProperty(PropertyName = "webviewBundleServingPrefix")]
    public string WebviewBundleServingPrefix { get; set; }
    [JsonProperty(PropertyName = "rootDir")]
    public string RootDir { get; set; }
    [JsonProperty(PropertyName = "injectScript")]
    public string InjectScript { get; set; }
    [JsonProperty(PropertyName = "injectStyle")]
    public string InjectStyle { get; set; }

    public enum ViewEnum
    {
      [EnumMember(Value = "multiple")]
      Multiple,
      [EnumMember(Value = "single")]
      Single,
    }
  }
}
